web/yo/app/scripts/services/pubCache.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/**
+ * @ngdoc factory
+ * @name oncokbApp.pubCache
+ * @author Jing Su
+ * @description
+ * This factory is a cache service used for storing validated PMIDs.
+ */
+angular.module('oncokbApp')
+    .factory('pubCache', function($q, FindRegex) {
+        var validatedPubs = {};
+
+        function clear() {
+            validatedPubs = {};
+        }
+        function get(pubs) {
+            var validatedPubsArray = [];
+            var notValidatedPubs = [];
+            _.each(pubs, function(pub){
+                if (validatedPubs[pub.id]) {
+                    validatedPubsArray.push(validatedPubs[pub.id]);
+                } else {
+                    notValidatedPubs.push(pub);
+                }
+            });
+            return {
+                validatedPubs: validatedPubsArray,
+                notValidatedPubs: notValidatedPubs
+            };
+        }
+        function set(pubs) {
+            _.each(pubs, function(pub){
+                validatedPubs[pub.id] = pub;
+            });
+        }
+        function validatePub(pubs) {
+            var deferred = $q.defer();
+            FindRegex.validation(pubs).then(function(result) {
```

Comment:
Don't you want to utilize the cache first for known pmids?

Reply:
Yes, I used the get() to load known pmids first and also get pmids that are not validated. Then I'll use validatePub() to validate those unknown pmids.

Reply:
Oh I see. Why not just include that get logic in the validatePub?

Reply:
I want to separate them to keep each function minimal.

Reply:
I would not worry about make the function minimal particularly  for this function. It's a util function anyway. But up to you.